packages/wrangler/e2e/dev.test.ts

```diff
@@ -733,3 +733,122 @@ describe("writes debug logs to hidden file", () => {
 		});
 	});
 });
+
+describe("zone selection", () => {
+	let worker: DevWorker;
+
+	beforeEach(async () => {
+		worker = await makeWorker();
+		await worker.seed((workerName) => ({
+			"wrangler.toml": dedent`
+					name = "${workerName}"
+					main = "src/index.ts"
+					compatibility_date = "2023-01-01"
+					compatibility_flags = ["nodejs_compat"]
+			`,
+			"src/index.ts": dedent`
+					export default {
+						fetch(request) {
+							return new Response(request.url)
+						}
+					}`,
+			"package.json": dedent`
+					{
+						"name": "${workerName}",
+						"version": "0.0.0",
+						"private": true
+					}
+					`,
+		}));
+	});
+
+	it("defaults to a workers.dev preview", async () => {
+		await worker.runDevSession("--remote --ip 127.0.0.1", async (session) => {
+			const { text } = await retry(
+				(s) => s.status !== 200,
+				async () => {
+					const r = await fetch(`http://127.0.0.1:${session.port}`);
+					return { text: await r.text(), status: r.status };
+				}
+			);
+			expect(text).toContain(`devprod-testing7928.workers.dev`);
```

Comment:
Yeah, this is definitely a tricky problem. For these tests we can't really make them account-agnostic because of the requirement to have a zone configured (as you mentioned), and in future I'd like to add some additional tests around Access and Wrangler, which will likely require even more specific account config.

Failing early for tests like these is a good shout though, I'll add that.